plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerSplit.java

```diff
@@ -15,5 +15,14 @@
 
 import io.trino.spi.connector.ConnectorSplit;
 
+import static com.google.common.base.Preconditions.checkArgument;
+
 public record FakerSplit(long splitNumber, long rowsCount)
-        implements ConnectorSplit {}
+        implements ConnectorSplit
+{
+    public FakerSplit
+    {
+        checkArgument(splitNumber >= 0, "splitNumber is negative");
+        checkArgument(rowsCount >= 0, "rowsCount is negative");
```

Comment:
> 0 ?

Reply:
Not sure - I think 0 may be needed for some cases.